courseware/api_test.py

```diff
@@ -71,3 +78,96 @@ def test_create_edx_user_conflict(user, scenario):
     create_edx_user(user)
 
     assert CoursewareUser.objects.count() == 0
+
+
+@responses.activate
+@freeze_time("2019-03-24 11:50:36")
+def test_create_edx_auth_token(settings, user):
+    """Tests create_edx_auth_token makes the expected incantations to create a OpenEdxApiAuth"""
+    refresh_token = "abc123"
+    access_token = "def456"
+    code = "ghi789"
+    responses.add(
+        responses.GET,
+        f"{settings.OPENEDX_API_BASE_URL}/auth/login/mitxpro-oauth2/?auth_entry=login",
+        status=status.HTTP_200_OK,
+    )
+    responses.add(
+        responses.GET,
+        f"{settings.OPENEDX_API_BASE_URL}/oauth2/authorize",
+        headers={
+            "Location": f"{settings.SITE_BASE_URL}/login/_private/complete?code={code}"
+        },
+        status=status.HTTP_302_FOUND,
+    )
+    responses.add(
+        responses.GET,
+        f"{settings.SITE_BASE_URL}/login/_private/complete",
+        status=status.HTTP_200_OK,
+    )
+    responses.add(
+        responses.POST,
+        f"{settings.OPENEDX_API_BASE_URL}/oauth2/access_token",
+        json=dict(
+            refresh_token=refresh_token, access_token=access_token, expires_in=3600
+        ),
+        status=status.HTTP_200_OK,
+    )
+
+    create_edx_auth_token(user)
+
+    assert len(responses.calls) == 4
+    assert dict(parse_qsl(responses.calls[3].request.body)) == dict(
+        code=code,
+        grant_type="authorization_code",
+        client_id=settings.OPENEDX_API_CLIENT_ID,
+        client_secret=settings.OPENEDX_API_CLIENT_SECRET,
+        redirect_uri=f"{settings.SITE_BASE_URL}/login/_private/complete",
+    )
+
+    assert OpenEdxApiAuth.objects.filter(user=user).exists()
+
+    auth = OpenEdxApiAuth.objects.get(user=user)
+
+    assert auth.refresh_token == refresh_token
+    assert auth.access_token == access_token
+    # plus one hour, minutes 10 seconds
+    assert auth.access_token_expires_on == datetime(
+        2019, 3, 24, 12, 50, 26, tzinfo=pytz.UTC
+    )
```

Comment:
Is it possible just to confirm this `access_token_expires_on` by using `timedelta` on the frozen time?